### PR TITLE
5903 Add support to set MLFlow `run_name` in MonaiAlgo

### LINKED ISSUE #5903: Experiment tracking with MonaiAlgo
**Is your feature request related to a problem? Please describe.**
Currently, the same `run_name` will be used if using "tracking" default setup shown in the [docstring](https://github.com/Project-MONAI/MONAI/blob/86c7ecdc1225529a379e2cd58c23fe53aa79b0d0/monai/fl/client/monai_algo.py#L359)

**Describe the solution you'd like**
`run_name` should be automatically defined based on client identity and experiment run (`job_id` in the nvflare case)

**Describe alternatives you've considered**
Manually, configuring the `run_name` argument of `MLFlowHandler` would be too cumbersome as it would need to change for each experiment and site.

**Additional context**
Manually setting the run_name for each client works but is cumbersome and error-prone.
image

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #5903 .

### Description

This PR added support to automatically set MLFlow `run_name` in the MonaiAlgo.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/fl/client/monai_algo.py [hunk 1]
```diff
@@ -465,6 +469,9 @@ def initialize(self, extra=None):
             self.eval_workflow.bundle_root = self.bundle_root
             if self.tracking is not None and isinstance(self.eval_workflow, ConfigWorkflow):
                 ConfigWorkflow.patch_bundle_tracking(parser=self.eval_workflow.parser, settings=settings_)
+                # set the MLFlow run name to the FL client name and timestamp
+                if self.eval_workflow.parser.get("run_name", None) is None:
+                    self.eval_workflow.parser["run_name"] = f"{self.client_name}_{time.strftime('%Y%m%d_%H%M%S')}"
```
Comment (holgerroth) [inline monai/fl/client/monai_algo.py:474]: We should compute the timestamp once and use it in both train and eval workflows.

**Review by holgerroth — APPROVED** [review]
Nice change. Confirmed with nvflare 2.3.0

File: monai/fl/client/monai_algo.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/fl/client/monai_algo.py]: Sure, updated in the latest commit.

Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Do you have any other comments?

Thanks in advance.

File: monai/fl/client/monai_algo.py [hunk 2]
```diff
@@ -357,15 +347,16 @@ def __init__(
         local_epochs: int = 1,
         send_weight_diff: bool = True,
         config_train_filename: str | list | None = "configs/train.json",
+        train_args: dict | None = None,
```
Comment (wyli) [inline monai/fl/client/monai_algo.py:350]: Should this be additional_train_args and move it to the end of the argument list?

Comment (Nic-Ma) [reply monai/fl/client/monai_algo.py:350]: Hi @wyli ,

I tried to be consistent with existing MONAI args, like:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/bundle/scripts.py#L1154.
@holgerroth What do you think?

Thanks.

Comment (Nic-Ma) [reply monai/fl/client/monai_algo.py]: Updated to "train_kwargs" now.

Thanks.

**Pull request merged**